nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,390 @@
+            InputStream is = response.body().byteStream();
+
+            Record record = handleResponse(is, coordinates);
+
+            return record != null
+                    ? Optional.of(record)
+                    : Optional.empty();
+        } catch (Exception e) {
+            getLogger().error("Could not execute lookup.", e);
+            throw new LookupFailureException(e);
+        }
+    }
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (propertyDescriptorName.startsWith("header")) {
+            String header = propertyDescriptorName.substring(propertyDescriptorName.indexOf(".") + 1);
+            return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .displayName(header)
+                .addValidator(Validator.VALID)
+                .dynamic(true)
+                .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+                .build();
+        }
+
+        return null;
+    }
+
+    protected Response executeRequest(Request request) throws IOException {
+        return client.newCall(request).execute();
+    }
+
+    private Record handleResponse(InputStream is, Map<String, Object> coordinates) throws SchemaNotFoundException, MalformedRecordException, IOException {
+        Map<String, String> variables = coordinates.entrySet().stream()
+            .collect(Collectors.toMap(
+                e -> e.getKey(),
+                e -> e.getValue().toString()
+            ));
+        try (RecordReader reader = readerFactory.createRecordReader(variables, is, getLogger())) {
+
+            Record record = reader.nextRecord();
+
+            if (recordPath != null) {
+                Optional<FieldValue> fv = recordPath.evaluate(record).getSelectedFields().findFirst();
+                if (fv.isPresent()) {
+                    FieldValue fieldValue = fv.get();
+                    RecordSchema schema = new SimpleRecordSchema(Arrays.asList(fieldValue.getField()));
+                    String[] parts = recordPath.getPath().split("/");
+                    String last = parts[parts.length - 1];
+
+                    Record temp;
+                    Object value = fieldValue.getValue();
+                    if (value instanceof Record) {
+                        temp = (Record) value;
+                    } else if (value instanceof Map) {
+                        temp = new MapRecord(schema, (Map<String, Object>) value);
+                    } else {
+                        temp = new MapRecord(schema, new HashMap<String, Object>() {{
+                            put(last, value);
+                        }});
+                    }
+
+                    record = temp;
+                }
+            }
+
+            return record;
+        } catch (Exception ex) {
+            is.close();
+            throw new RuntimeException(ex);
```

Comment:
Would avoid wrapping checked exceptions that are known to be thrown by method (such as IOException) in a RuntimeException. Am guessing you can just use `throw ex` because all checked exceptions that can be caught are likely already declared for the method but not sure off the top of my head.

Reply:
Done.